custom_model_runner/datarobot_drum/drum/adapters/model_adapters/python_model_adapter.py

```diff
@@ -103,14 +105,55 @@ def __init__(self, model_dir, target_type=None):
         # New custom task class and instance loaded from custom.py
         self._custom_task_class = None
         self._custom_task_class_instance = None
+        self._guard_moderation_hooks = None
+        self._guard_pipeline = None
 
         if target_type == TargetType.TEXT_GENERATION:
             self._target_name = os.environ.get("TARGET_NAME")
             if not self._target_name:
                 raise ValueError("Unexpected empty target name for text generation!")
+            self._load_guard_hooks_for_drum()
         else:
             self._target_name = None
 
+    def _load_guard_hooks_for_drum(self):
+        if os.environ.get("MLOPS_GUARD_HOOK_FILE", None) is None:
+            return
+
+        guard_file_prefix = os.environ.get("MLOPS_GUARD_HOOK_FILE")
```

Comment:
Could you please do some refactoring and unify common code for the regular hooks loading and guards hook loading?

Reply:
I will try to see if I can, didn't want to break the stuff which is already running.